src/config/builder.rs

```diff
@@ -60,8 +61,8 @@ impl From<Config> for ConfigBuilder {
 }
 
 impl ConfigBuilder {
-    pub fn build(self) -> Result<Config, Vec<String>> {
-        let (config, warnings) = self.build_with_warnings()?;
+    pub fn build(self, pipelines: Pipelines) -> Result<Config, Vec<String>> {
```

Comment:
A lot of the diff seems related to this change, and I'm not sure it's the best path. `ConfigBuilder` is meant to represent a whole configuration, so why not one of these two strategies?

1. Add a `pipelines` field to `ConfigBuilder` that we can populate before calling `build`
2. Add an `add_pipeline` function to `ConfigBuilder` that accepts a name and transform, namespaces the transform names, and adds them to the existing `transforms` field normally

These should result in a simpler change overall and better align with the builder pattern we're using here.

Reply:
👍 I'll do that, but I'll try to merge [this PR](https://github.com/timberio/vector/pull/8706) first to have the `PipelineId` in place and scope every transform.

Reply:
I did it here https://github.com/timberio/vector/pull/8810 because `add_pipeline`, if it adds the transforms, implies a verification step which is done when converting from `ConfigBuilder` to `Config`.
So to avoid that, the pipelines are just added to the `ConfigBuilder` as they are and converted after.